[Publishing] Adds image component and font system

kanaabe opened the pull request:
Thanks @eessex for pairing on this & getting me started in reaction-force! Main thing I changed since the pair sesh is how the fonts work. I tried making it feel more similar to Force/Positron so whenever a font needs to be used, we can `${Fonts.garamond("s15")}` in the styled component.

This screenshot doesn't show the new font since it's not in my system yet, but the fallback is good old Arial. 

![image](https://user-images.githubusercontent.com/2236794/27661907-05958552-5c2a-11e7-8c5f-750923b57b96.png)

eessex commented on src/components/publishing/artwork.tsx:
```diff
@@ -1,14 +1,14 @@
 
 const TruncatedLine = styled.div`
+  ${Fonts.garamond("s15")}
   display: block;
```
this is great!

alloy replied:
Agreed ✨

eessex commented on src/components/publishing/image.tsx:
```diff
@@ -0,0 +1,46 @@
+  margin: 0;
+  border-bottom: 1px solid black;
+  cursor: pointer;
+  display: inline-block;
```
might be a good idea to add padding left here, so that long captions don't sit too close

eessex commented on src/components/publishing/image.tsx:
```diff
@@ -0,0 +1,46 @@
+
+const BlockImage = styled.img`
+  display: block;
+`
```
what does the display do here?

orta replied:
`img` is typically an inline element, I think, so this tells it to "move to a new line"

kanaabe replied:
What he said 😺 . Without it the inline version gives us a ghost space of something like 3px

alloy commented on src/components/publishing/fonts.tsx:
```diff
@@ -0,0 +1,110 @@
+  const evaluatedSize = textSizesForUnica[size]
+  const style = css`
+    font-family: ${unicaFontFamily};
+    -webkit-font-smoothing: antialiased;
```
Are you prefixing this yourself because you want it to _only_ apply to webkit? Otherwise styled-components should auto-prefix it.

kanaabe replied:
Hmm interesting point. I was mostly following https://github.com/artsy/reaction-force/blob/master/src/assets/fonts.tsx#L31 but it looks like the prefix doesn't get applied if I use `font-smooth: antialiased` or `font-smoothing: antialiased`.

alloy replied:
Oh yeah indeed, I remember asking @l2succes the same thing :D

alloy commented on src/components/publishing/image.tsx:
```diff
@@ -0,0 +1,46 @@
+  display: block;
+`
+
+class Image extends Component<any, null> {
```
Prefer `void` over `null` for stateless components.

alloy replied:
Btw, in the case of stateless components you can also choose for just [a function](https://facebook.github.io/react/docs/components-and-props.html#functional-and-class-components).

alloy commented on src/components/publishing/image.tsx:
```diff
@@ -0,0 +1,46 @@
+    const { image } = this.props
+    return (
+      <div className="article-image">
+        <BlockImage src={image.url} width={"100%"} />
```
The curly brackets aren’t strictly needed when specifying a static string.

alloy commented:
No blockers from my side, looks great! 🚀

kanaabe commented:
Thanks for the comments! I simplified the image component to a function (good call this will likely never have a state) and added a long caption image to the storybook. Will merge when 🍏
